textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -32,78 +32,86 @@
-                luceneFieldType.setStored(true);
-                luceneFieldType.setStoreTermVectors( true );
-                luceneFieldType.setStoreTermVectorOffsets( true );
-                luceneFieldType.setStoreTermVectorPayloads( true );
-                luceneFieldType.setStoreTermVectorPositions( true );
-                luceneFieldType.setTokenized( true );
-
-                luceneField = new org.apache.lucene.document.Field(
-                        fieldName,(String) fieldValue,luceneFieldType);
-
-                break;
+        switch (fieldType) {
+        case STRING:
+            luceneField = new org.apache.lucene.document.StringField(fieldName, (String) fieldValue, Store.YES);
+            break;
+        case INTEGER:
+            luceneField = new org.apache.lucene.document.IntField(fieldName, (Integer) fieldValue, Store.YES);
+            break;
+        case DOUBLE:
+            double value = (Double) fieldValue;
+            luceneField = new org.apache.lucene.document.DoubleField(fieldName, value, Store.YES);
+            break;
+        case DATE:
+            String dateString = DateTools.dateToString((Date) fieldValue, Resolution.MILLISECOND);
+            luceneField = new org.apache.lucene.document.StringField(fieldName, dateString, Store.YES);
+            break;
+        case TEXT:
+            // By default we enable positional indexing in Lucene so that we can
+            // return
+            // information about character offsets and token offsets
+            org.apache.lucene.document.FieldType luceneFieldType = new org.apache.lucene.document.FieldType();
+            luceneFieldType.setIndexOptions(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS);
+            luceneFieldType.setStored(true);
+            luceneFieldType.setStoreTermVectors(true);
+            luceneFieldType.setStoreTermVectorOffsets(true);
+            luceneFieldType.setStoreTermVectorPayloads(true);
+            luceneFieldType.setStoreTermVectorPositions(true);
+            luceneFieldType.setTokenized(true);
+
+            luceneField = new org.apache.lucene.document.Field(fieldName, (String) fieldValue, luceneFieldType);
+
+            break;
 
         }
         return luceneField;
     }
+
     /**
-     * @about Creating a new span tuple from span schema, field list 
+     * @about Creating a new span tuple from span schema, field list
      */
-    public static ITuple getSpanTuple( List<IField> fieldList, List<Span> spanList, Schema spanSchema) {
+    public static ITuple getSpanTuple(List<IField> fieldList, List<Span> spanList, Schema spanSchema,
+            boolean removeLastField) {
         IField spanListField = new ListField<Span>(new ArrayList<>(spanList));
         List<IField> fieldListDuplicate = new ArrayList<>(fieldList);
+        // If SourceOperator is of type Keyword, KeywordMatcher also includes a
+        // span which should be discarded by dictionary matcher.

```

Comment:
We only need to remove the last field because the last field contains the list of spans. So, even if it is "New York" or "new york city", the last field will be the list of spans. For "new york" it'll be a list of size 2, for "new york city" it'll be a list of size 3. When we delete that field, we delete the list altogether, and create a new span.